content/backend/graphql-scala/2-preparing-first-query.md

```diff
@@ -0,0 +1,377 @@
+      query, // 11
+      MyContext(dao), // 12
+      variables = vars, // 13
+      operationName = operation // 14
+    ).map(OK -> _)
+      .recover {
+      case error: QueryAnalysisError => BadRequest -> error.resolveError
+      case error: ErrorWithResolver => InternalServerError -> error.resolveError
+    }
+  }
+
+}
+
+```
+
+</Instruction>
+
+It's one of the most important files in entire backend server so let's analyze it step by step:
+* **1** We need access to the database, so it's the step where it's created.
+* **2** `endpoint` responds with `Route` type, so it will be used directly in the routing of HTTP server. It expects JSON object as parameter.
+* **3** There `fields` JsObject is extracted from the root object. The format of root JSON object expected by GraphQL implementation looks like this:
+
+  ```
+  {
+  query: {},
+  variables: {},
+  operationName: ""    
+  }
+  ```
+
+  `query` is a query itself, `variables` is additional data for that query. In GraphQL you can send the query and arguments separately. You can also set name for the query, it’s what the third object is for. Imagine that query is like a function, usually you’re using anonymous functions, but for logging or other purposes you could add names. It’s sent as `operationName`.
+
+* **4** We're extracting `query` from request at this point.
+* When we have the query, we have to parse it. Sangria provides `QueryParser.parse` (**5**) function we can use in this case. In case of failure, we're responding with status 400 and error description. In case of successful parsing, we're also trying to extract also `operationName`(**6**) and `variables`(**7**) and all those three object we're passing to the execution function (**8**).
+* **9** `Executor.execute` is the most important call in this class because it's the point where query is executed. If executor responds with success, the result is send back to the client, in the other case server will responds with status code 4xx and some kind of explanation what was wrong with the query. Executor needs some data to fulfil the request. Three of them are `query`(**11**), `operationName`(**13**) and `variables`(**14**) which are read from request.
+The last two are: `GraphQLSchema.SchemaDefinition` and `MyContext(dao)`.
+* **12** MyContext is a context object mentioned above. In this case we're building the context object with prepared DAO class.
+* `GraphQLSchema.SchemaDefinition` is the last object we have to explain here. It contains our Schema - what we are able to query for. It also interprets how data is fetched and from which data source (i.e. one or more databases, REST call to the other server…). In short our `SchemaDefinition` file defines what we want to expose. There are defined types (from GraphQL point of view) and shape of the schema a client is able to query for.
+
+### Define GraphQLSchema
+
+<Instruction>
+
+Create GraphQLSchema object:
+
+```scala
+import com.howtographql.scala.sangria.models.Link
```

Comment:
This package doesn't exist

Reply:
Up to this point no file had a `package` - everything was thrown into the main dir